src/components/Clients/ClientListTableRow.jsx

```diff
@@ -65,45 +62,55 @@ const ClientListTableRow = ({ labelId, client, state, dispatch }) => {
-          inputProps={{
-            'aria-labelledby': labelId
-          }}
-          onClick={() => {
-            handleSelectClient(client);
-          }}
-        />
-      </StyledTableCell>
-      <StyledTableCell align="center">{client.person}</StyledTableCell>
-      {/* ***** TODO: Switch this webId to being a small Notes section */}
-      {/* seems having a link or even displaying another user's pod is completely useless/irrelevant */}
-      {/* see no reason it would ever need to be used, but notes/comments will be of utmost importance to caseworkers */}
-      <StyledTableCell align="center">
-        <Link
-          to={client.webId}
-          target="_blank"
-          rel="noreferrer"
-          style={{ textDecoration: 'none', color: theme.palette.primary.dark }}
-        >
-          Link to Pod Profile
-        </Link>
-      </StyledTableCell>
-      <StyledTableCell align="center">
-        <IconButton size="large" edge="end" onClick={handlePinClick}>
-          {pinnedIcon}
-        </IconButton>
-      </StyledTableCell>
-      <StyledTableCell align="center">
-        <IconButton size="large" edge="end" onClick={() => handleDeleteClient()}>
-          <DeleteOutlineOutlinedIcon />
-        </IconButton>
-      </StyledTableCell>
-    </StyledTableRow>
+    <>
+      <StyledTableRow>
+        <StyledTableCell align="center">
+          <Checkbox
+            id={labelId}
+            color="primary"
+            checked={selectedUser.webId === client.webId}
+            inputProps={{
+              'aria-labelledby': labelId
+            }}
+            onClick={() => {
+              handleSelectClient(client);
+            }}
+          />
+        </StyledTableCell>
+        <StyledTableCell align="center">{client.person}</StyledTableCell>
+        {/* ***** TODO: Switch this webId to being a small Notes section */}
+        {/* seems having a link or even displaying another user's pod is completely useless/irrelevant */}
+        {/* see no reason it would ever need to be used, but notes/comments will be of utmost importance to caseworkers */}
+        <StyledTableCell align="center">
+          <Link
+            to={client.webId}
+            target="_blank"
+            rel="noreferrer"
+            style={{ textDecoration: 'none', color: theme.palette.primary.dark }}
+          >
+            Link to Pod Profile
+          </Link>
+        </StyledTableCell>
+        <StyledTableCell align="center">
+          <IconButton size="large" edge="end" onClick={handlePinClick}>
+            {pinnedIcon}
+          </IconButton>
+        </StyledTableCell>
+        <StyledTableCell align="center">
+          <IconButton size="large" edge="end" onClick={() => setShowClientDeleteModal(true)}>
+            <DeleteOutlineOutlinedIcon />
+          </IconButton>
+        </StyledTableCell>
+      </StyledTableRow>
+
+      {/* modal/popup renders when showDeleteClientModal state is true */}
+      <DeleteClientModal
```

Comment:
It makes more sense to me to put this on the Table or Page level. That way we won't be rendering 100 different modals for 100 different rows. It would also resolve the `React.Fragment` issue that Andy mentioned higher up.

You can then have the `handleDeleteClient` function call some passed-in messaging function that triggers the modal, like 

`const handleDeleteClient = () => props.callDelete(client)`

You could also use a reducer, and have the delete fire off an action, but that will be hard to integrate with the context. I haven't really thought about how to integrate contexts with reducers, so I don't have much advice there.